heimdall-frontend/source/mainwindow.cpp

```diff
@@ -659,7 +659,7 @@ void MainWindow::SelectPartitionName(int index)
 
 		partitionFileGroup->setTitle(title);
 
-		if (!fileInfo.GetFilename().isEmpty())
+		if (pitEntry && !fileInfo.GetFilename().isEmpty())
```

Comment:
Fair enough. I imagine the original check was just me being a lazy copy 'n' paste programmer - which was the cause of most the other bugs you've fixed too. Whoops!